test/run_envoy_tests.sh

```diff
@@ -20,15 +20,33 @@ fi
 : ${TEST_TMPDIR:=$TEST_BASE/tmp}
 : ${TEST_SRCDIR:=$TEST_BASE/runfiles}
 export TEST_TMPDIR TEST_SRCDIR
+export TEST_WORKSPACE=""
+export TEST_UDSDIR="$TEST_TMPDIR"
 
 echo "TEST_TMPDIR=$TEST_TMPDIR"
 echo "TEST_SRCDIR=$TEST_SRCDIR"
 
 mkdir -p $TEST_TMPDIR
 
 $SOURCE_DIR/test/certs/gen_test_certs.sh $TEST_SRCDIR/test/certs
-$SOURCE_DIR/test/config/integration/gen_test_configs.sh $SOURCE_DIR/test/config/integration \
-  $TEST_SRCDIR/test/config/integration
+
+# Some hacks for the config file template substitution. These go away in the Bazel build.
+CONFIG_IN_DIR="$SOURCE_DIR"/test/config/integration
+CONFIG_RUNFILES_DIR="$TEST_SRCDIR/$TEST_WORKSPACE"/test/config/integration
+CONFIG_OUT_DIR="$TEST_TMPDIR"/test/config/integration
+mkdir -p "$CONFIG_RUNFILES_DIR"
+mkdir -p "$CONFIG_OUT_DIR"
+cp "$CONFIG_IN_DIR"/*.json "$CONFIG_RUNFILES_DIR"
```

Comment:
Under Bazel, runfiles are where the source dependencies and pregenerated (from other rules) dependencies live. They are read-only in the build/test sandbox. The tmpdir is where we can do writable stuff.

Unfortunately, we can't pregenerate the JSON configs for the tests and put them in runfiles, since they may have template variables that refer to tmpdir, which is only known at test execution time (not build time). So, we need to put them in tmpdir. 

The Bazel `environment_sub.sh` rule will take the JSON templates from runfiles and generate in tmpdir (see #642). With the cmake build, we actually need to copy from source to runfiles first to make this transparent to scripts like `environment_sub.sh`. This disappears soon when we remove cmake.